scripts/build-renode.sh

```diff
@@ -75,12 +76,30 @@ RENODE_RESC="$RENODE_SCRIPTS_DIR/litex_buildenv.resc"
 RENODE_REPL="$RENODE_SCRIPTS_DIR/litex_buildenv.repl"
 
 mkdir -p $RENODE_SCRIPTS_DIR
-python $TOP_DIR/third_party/litex-renode/generate-renode-scripts.py $LITEX_CONFIG_FILE \
-	--repl "$RENODE_REPL" \
-	--resc "$RENODE_RESC" \
-	--bios-binary "$TARGET_BUILD_DIR/software/bios/bios.bin" \
-	--firmware-binary "$TARGET_BUILD_DIR/software/$FIRMWARE/firmware.bin" \
-	--configure-network ${TAP_INTERFACE:-""}
+
+KERNEL_FLASH_OFFSET=`python $LITEX_RENODE/litex-config-parser.py $LITEX_CONFIG_FILE --constant kernel_image_flash_offset`
+ROOTFS_FLASH_OFFSET=`python $LITEX_RENODE/litex-config-parser.py $LITEX_CONFIG_FILE --constant rootfs_image_flash_offset`
+DTB_FLASH_OFFSET=`python $LITEX_RENODE/litex-config-parser.py $LITEX_CONFIG_FILE --constant device_tree_image_flash_offset`
+EMULATOR_FLASH_OFFSET=`python $LITEX_RENODE/litex-config-parser.py $LITEX_CONFIG_FILE --constant emulator_image_flash_offset`
+
+if [ "$FIRMWARE" == "linux" ]; then
+	python $LITEX_RENODE/generate-renode-scripts.py $LITEX_CONFIG_FILE \
```

Comment:
Rather than doing this parsing in bash / shell, could we change `generate-renode-scripts.py` to understand the flash offsets? IE Something like;
`--flash-binary "KERNEL_FLASH_OFFSET:$TARGET_BUILD_DIR/software/linux/firmware.bin"` and then `generate-renode-scripts.py` looks up the `KERNEL_FLASH_OFFSET` value in the csr output?

Reply:
My idea was to make `--flash-binary` switch generic -- i.e., allow to load any binary under any offset. This, however, required to know the offset beforehand - hence the need of the second parser. I agree that this probably increases the complication too much.

I agree with your suggestion - I will keep the current syntax for `--flash-binary`, i.e., `--flash-binary PATH:OFFSET`, but allow `OFFSET` to be either a number or one of predefined text labels (like `KERNEL_FLASH_OFFSET`) with values parsed from the `csr.csv` configuration file.

Reply:
I created a PR with the suggested changes to `generate-renode-scripts.py`: https://github.com/litex-hub/litex-renode/pull/23.

Reply:
using labels parsed by `generate-renode-scripts.py` now